error: use of undeclared identifier 'NSIG'

### Description of the problem

Hi, I'm the maintainer of MyPaint in MacPorts, finally getting around to updating to version 2.

MyPaint 2.0.1 failed to build for me. The error was:

```
In file included from lib/fill/fill_common.cpp:10:
In file included from lib/fill/fill_common.hpp:26:
In file included from /opt/local/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/numpy/core/include/numpy/arrayobject.h:5:
In file included from /opt/local/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/numpy/core/include/numpy/npy_interrupt.h:84:
/usr/include/signal.h:69:42: error: use of undeclared identifier 'NSIG'
extern __const char *__const sys_signame[NSIG];
                                         ^
/usr/include/signal.h:70:42: error: use of undeclared identifier 'NSIG'
extern __const char *__const sys_siglist[NSIG];
                                         ^
1 warning and 2 errors generated.
```

I believe this is occurring because you are defining `_POSIX_C_SOURCE` but you are not also defining `_DARWIN_C_SOURCE`.

Lines 68-71 of /usr/include/signal.h read as follows:

```
#if !defined(_ANSI_SOURCE) && (!defined(_POSIX_C_SOURCE) || defined(_DARWIN_C_SOURCE))
extern __const char *__const sys_signame[NSIG];
extern __const char *__const sys_siglist[NSIG];
#endif
```

This fix worked for me:

```
--- setup.py.orig	2020-05-29 11:40:02.000000000 -0500
+++ setup.py	2020-06-02 21:15:33.000000000 -0500
@@ -874,7 +874,7 @@
     extra_link_args = []
 
     if sys.platform == "darwin":
-        pass
+        extra_compile_args.append('-D_DARWIN_C_SOURCE')
     elif sys.platform == "win32":
         pass
     elif sys.platform == "msys":
```

#### Basic system details

MyPaint version: 2.0.1
Operating system: macOS 10.13.6
Desktop environment: X11

#### Steps to reproduce

Build MyPaint

--- a/setup.py
+++ b/setup.py
@@ -874,7 +874,7 @@ def get_ext_modules():
     extra_link_args = []
 
     if sys.platform == "darwin":
-        pass
+        extra_compile_args.append('-D_DARWIN_C_SOURCE')
     elif sys.platform == "win32":
         pass
     elif sys.platform == "msys":

